### PR TITLE
feat: [CI-4442]: handle singe callback

### PR DESCRIPTION
**Pull request by harshjain12**
**1.   Handle single callback.
2.  Abort parked task if lite engine fails to execution
3.  Wrap hint and explanation.
4.  Support aborting parked task.
5.  Notify ci manager in case of task abort.**

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33107)

### CODE REVIEW

**Comment by harshjain12** [comment]
trigger messagecheck

**Comment by harshjain12** [comment]
trigger messagecheck

**Review by raghavharness — APPROVED** [review]

**Review by raghavharness — APPROVED** [review]

**Comment by harshjain12** [comment]
trigger ft2

File: 320-ci-execution/src/main/java/io/harness/states/AbstractStepExecutable.java [hunk 1]
```diff
@@ -172,6 +205,17 @@ public List<TaskSelector> fetchDelegateSelector(Ambiance ambiance) {
     return connectorUtils.fetchDelegateSelector(ambiance, executionSweepingOutputResolver);
   }
 
+  private void abortTasks(List<String> allCallbackIds, String callbackId, Ambiance ambiance) {
+    List<String> callBackIds =
+        allCallbackIds.stream().filter(cid -> !cid.equals(callbackId)).collect(Collectors.toList());
+    callBackIds.forEach(callbackId1 -> {
+      waitNotifyEngine.doneWith(callbackId1,
+          ErrorNotifyResponseData.builder()
+              .errorMessage("Delegate is not able to connect to created build farm")
```
Comment (shubham149) [inline 320-ci-execution/src/main/java/io/harness/states/AbstractStepExecutable.java:214]: This error message is only valid for K8 and not for VMs.

**Comment by hextechpal** [comment]
*[`320-ci-execution/src/main/java/io/harness/states/AbstractStepExecutable.java` line 344 at r2](https://reviewable.io/reviews/harness/harness-core/33107#-N2AHE7kBOEcF1a39-9E:-N2AHE7kBOEcF1a39-9F:b-voclz0) ([raw file](https://github.com/harness/harness-core/blob/5db9ff87e93dd5482bf7dcf46861f5fbbd2b254f/320-ci-execution/src/main/java/io/harness/states/AbstractStepExecutable.java#L344)):*
> ```Java
> 
>         return StepResponse.builder()
>             .status(Status.ERRORED)
> ```

@harshjain12 can you change this to failed instead of errored. Errored is internal pipeline status, I will remove from contract

**Pull request merged**